oidc.go

```diff
@@ -174,10 +174,10 @@ func (o *OIDCClient) ExchangeWithNonce(code, nonce string, options map[string]st
 }
 
 // Returns the URL for authorization request.
-func (o *OIDCClient) AuthRequestURL(state string, options map[string]string) (string, error) {
+func (o *OIDCClient) AuthRequestURL(state string, options map[string]interface{}) (string, error) {
 	var opts []oauth2.AuthCodeOption
 	for key, value := range options {
-		opts = append(opts, oauth2.SetAuthURLParam(key, value))
+		opts = append(opts, oauth2.SetAuthURLParam(key, value.(string)))
 	}
 	return o.oauth2Config.AuthCodeURL(state, opts...), nil
```

Comment:
Assuming that the `options` argument is defined as follows (some mandatory parameters are probably missing but the `login_hint` part is the important one).

```go
options := map[string]interface{}{
    "login_hint": []string{"birthdate-ddmmyy", "phone-nnnnnnnn"},
    "ui_locales": "nb",
    "scope": "openid profile signicat.sign",
    "signicat_signtext": "I confirm my purchase of broadband subscription Medium500.",
    "acr_values": "urn:signicat:oidc:method:nbid-inapp-sign"
}
```
This would break as string slice cannot be casted to string.

Something like this would handle the conversion
```suggestion
	var opts []oauth2.AuthCodeOption
	var err error = nil
	for key, value := range options {
		switch valueType := value.(type) {
		case []string:
			for _, item := range value.([]string) {
				opts = append(opts, oauth2.SetAuthURLParam(key, item))
			}
		case string:
			opts = append(opts, oauth2.SetAuthURLParam(key, value.(string)))
		default:
			err = fmt.Errorf("unknown request option type: '%T'", valueType)
		}
	}
	return o.oauth2Config.AuthCodeURL(state, opts...), err
```

but the resulting URL would only contain the last value for `login_hint` given that `oauth2.SetAuthURLParam` replaces any existing values for a key with a new one. Based on [Signicat's documentation](https://developer.signicat.com/documentation/authentication/protocols/openid-connect/endpoints/) that's not going to work. We could build the URL manually but that's slightly more work which I'm not going to ask you to do unless you want to?

Reply:
@haaja Thanks for your quick response. Here comes my slightly slower response :)

So the question here is whether or not we should actually handle this case at all in the non-encrypted flow. I haven't seen any examples of where you would need to input complex objects at all into the normal flow. So should the code just push back with an error instead of trying to handle the object at all?

However I changed your proposal a bit by just joining the string slice together like the oauth implementation does with e.g. scopes.

Reply:
> So the question here is whether or not we should actually handle this case at all in the non-encrypted flow. I haven't seen any examples of where you would need to input complex objects at all into the normal flow. So should the code just push back with an error instead of trying to handle the object at all?

That's a valid question. I've never used complex objects with normal flow either but if my memory serves me right there at least used to be a way to use `login_hint` to prefill a phone number in one of the authentication methods in Finland. Unfortunately I don't have access to the documentation at the moment so I might just as well remember incorrectly. 

> However I changed your proposal a bit by just joining the string slice together like the oauth implementation does with e.g. scopes.

That's even better. It's probably still not enough for Signicat according to their documentation but that's fine. It's not like there's a standard on how to pass array values as a query parameter.